index.js

```diff
@@ -219,90 +222,98 @@ class ThreadStream extends EventEmitter {
     return true
   }
 
-  _hasSpace () {
-    const current = Atomics.load(this._state, WRITE_INDEX)
-    return this._data.length - this.buf.length - current > 0
-  }
-
   write (data) {
-    if (this.closed) {
+    if (this.destroyed) {
       throw new Error('the worker has exited')
     }
 
-    if (this.flushing && this.buf.length + data.length >= MAX_STRING) {
-      // process._rawDebug('write: flushing')
-      this._writeSync()
-      this.flushing = true // we are still flushing
+    if (this.ending) {
+      throw new Error('the worker is ending')
     }
 
-    if (!this.ready || this.flushing) {
-      this.buf += data
-      return this._hasSpace()
+    if (this.buf.length + data.length >= MAX_STRING) {
+      try {
+        this._writeSync()
```

Comment:
```suggestion
        this._writeSync()
        this.flushing = true
```

I think this is the problem